Update for SDK 2.2.0

ndabas opened the pull request.

ndabas committed:
Add support for SDK v2.0.0

ndabas committed:
Remove x86 build

ndabas committed:
Update packages

ndabas committed:
Remove COMSPEC

Fixes https://github.com/raspberrypi/pico-setup-windows/issues/36

ndabas committed:
Update VS Code config

ndabas committed:
pico_w httpd example builds again

Fixed in https://github.com/raspberrypi/pico-sdk/pull/1841

ndabas committed:
Update actions

ndabas committed:
Test riscv-gnu-toolchain build

ndabas committed:
Use shorter base path for builds

ndabas committed:
Clean up scripts

ndabas committed:
Don't fail if no DLLs to copy

ndabas committed:
Pre-archive RISCV toolchain

NSIS cannot handle content > 2GB

ndabas committed:
Update Arm GNU Toolchain

ndabas committed:
Don't wait for dangling processes

arm-gnu-toolchain launches a shell

ndabas committed:
Use actual path to toolchain

ndabas committed:
This should fail

ndabas committed:
Remove trailing backslash from toolchain path

neurothoughtmachine commented:
Greetings everyone,

After the initial install of the windows pico toolchain,

I am getting a write error during the cloning process from the windows shell "Can't find pico-v1.5.1".

Should I be using a different path than my user home path?

<img width="980" height="510" alt="image" src="https://github.com/user-attachments/assets/6735084b-7ad2-44f6-8ebc-cf969d54c3db" />

ndabas committed:
Update for Pico SDK 2.2.0

- Update repos and samples to SDK 2.2.0
- Update crawler to spoof User-Agent for developer.arm.com
- Update tools
- Configure make job slots more sensibly
- Add zcmp to RISC-V toolchain
- Strip binaries in RISC-V toolchain

ndabas committed:
Remove riscv-gnu-toolchain archive

ndabas committed:
Map SDK to shorten build paths

ndabas committed:
Update actions versions

ndabas committed:
Update Windows runner version

ndabas committed:
Shorten install paths for tests

ndabas committed:
Remove CMake packages reg key

CMake finds the package in the PATH anyway

ndabas committed:
Don't write to the registry

Additionally, don't package pico-{examples,extras,playground} with the installer

ndabas committed:
Remove VS Code

ndabas committed:
Remove Raspberry Pi branding

ndabas committed:
Shorten default install path

ndabas committed:
Remove pandoc

ndabas committed:
Separate out compile step

ndabas committed:
There is no matrix

ndabas committed:
Fix installer README path

ndabas committed:
Separate Windows Terminal profile logic

ndabas committed:
Disable char8_t because GCC libcody doesn't like it

ndabas committed:
Group logs in GitHub Actions

ndabas committed:
Some more log grouping

ndabas committed:
Refactor NSIS generator code

ndabas committed:
More log grouping

ndabas committed:
Add builds from config

ndabas committed:
Generate uninstaller sections as well

ndabas committed:
Fix test script to not require PowerShell 7

ndabas committed:
Set PICO_INSTALL_PATH for extended tests

ndabas committed:
Fix install dir constant

ndabas committed:
Map compiled outputs

ndabas committed:
Clean up build target selection

ndabas committed:
Remove version.ini

ndabas committed:
DRY file handling

ndabas committed:
Add zip generation

ndabas committed:
Archive compiled outputs separately as well

ndabas committed:
Add release automation

ndabas committed:
Update events

- Removed pull_request because the workflow runs on PR branch pushes anyway
- Added a catch-all branches because it didn't run on branches otherwise, with the tags-ignore.

ndabas committed:
Download all artifacts to the same dir

ndabas committed:
Add release notes

ndabas committed:
Fixes for env/setup scripts

- Make pico-setup.cmd standalone (no parameter needed for repo dir)
- Check OpenOCD and RISC-V binaries as well, with a warning if they are not found
- Remove OPENOCD_SCRIPTS env var as OpenOCD can find its own scripts now

ndabas committed:
Better compression for zip files

ndabas committed:
Update README

ndabas committed:
Update user README

ndabas committed:
Add Windows arm64 test

ndabas committed:
Disable Windows arm64 tests for now

ndabas committed:
Clarify Windows 11 ARM Docker status

ndabas committed:
Remove custom exec function

With $PSNativeCommandUseErrorActionPreference = $true, we don't need to check $LASTEXITCODE manually.

ndabas committed:
Add install types (full/typical/minimal)

ndabas committed:
Fix archive contents
